packages/venia-concept/src/actions/cart/__tests__/asyncActions.spec.js

```diff
@@ -897,6 +897,36 @@ describe('toggleCart', () => {
         expect(dispatch).not.toHaveBeenCalled();
     });
 
+    test('its thunk opens the drawer when cart is empty', async () => {
+        getState.mockReturnValueOnce({
+            app: { drawer: 'cart' },
+            cart: { details: { items_count: 0 } }
+        });
+        await toggleCart()(...thunkArgs);
+        // toggleDrawer
+        expect(dispatch).toHaveBeenNthCalledWith(1, expect.any(Function));
+        expect(dispatch).toHaveBeenCalledTimes(1);
+    });
+
+    test('its thunk opens the drawer when the cart has no details', async () => {
+        getState.mockReturnValueOnce({ app: { drawer: 'cart' }, cart: {} });
+        await toggleCart()(...thunkArgs);
+        // toggleDrawer
+        expect(dispatch).toHaveBeenNthCalledWith(1, expect.any(Function));
+        expect(dispatch).toHaveBeenCalledTimes(1);
+    });
+
+    test('its thunk opens the drawer when the cart has details', async () => {
+        getState.mockReturnValueOnce({
+            app: { drawer: 'cart' },
+            cart: { details: { items_count: 2 } }
+        });
+        await toggleCart()(...thunkArgs);
+        // toggleDrawer
+        expect(dispatch).toHaveBeenNthCalledWith(1, expect.any(Function));
+        expect(dispatch).toHaveBeenCalledTimes(1);
```

Comment:
Shouldn't `dispatch` be called twice here? First to `toggleDrawer` and then to get the details?

Reply:
I've tried to fixed it.please review and let me know.

Reply:
Hi @vishal-7037, I found the issue:

Lines 400-402 in `cart/asyncActions.js` are as follows:

```js
        if (app.drawer === 'cart') {
            return dispatch(closeDrawer());
        }
```

In each of these tests we are mocking `app.drawer` to equal "cart", so this conditional is getting hit and dispatching a `closeDrawer` action.

1. Please update these tests to set `app.drawer` to something other than "cart" (`null` would work).
2. Please update this specific test, as we expect `dispatch` to be called twice (`toggleDrawer` and `getCartDetails`).

Thanks!